src/python/pants/option/option_types.py

```diff
@@ -216,7 +216,7 @@ def __new__(
         cls,
         flag_name: str | None = None,
         *,
-        default: _MaybeDynamicT[list[_ListMemberT]] = [],
```

Comment:
Discovered there is already a check on line 234 to put in `[]` if `default` is falsy. Seems like making the default here `Optional` and passing in `None` is preferable to using the mutable default argument. Happy to back it out if we don't want to change the signature here.

Reply:
Agreed, since we can't get the type check to work anyway, better to not do the bad thing with the mutable default for no purpose.